src/gui/file_loader.py

```
@@ -0,0 +1,154 @@
+        filename = temp_path.name
+        match = uuid_pattern.match(filename)
+        original_filename = match.group(1) if match else filename
+        
+        # Create persistent path with UUID to avoid collisions
+        safe_name = f"{uuid.uuid4().hex}_{original_filename}"
+        dest_path = uploads_dir / safe_name
+        
+        try:
+            shutil.copy2(temp_path, dest_path)
+            persistent_files.append(str(dest_path.absolute()))
+        except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)